plugins/content/caption/base/CaptionPlugin.php

```diff
@@ -644,6 +655,69 @@ public static function getManifestEditors ($config)
 		return $contributors;
 	}
 
+	protected static function getLiveCaptionArray($entry, $deliveryProfile): array
+	{
+		$entryStreams = $entry->getStreams();
+		if(!count($entryStreams))
+		{
+			return array();
+		}
+
+		$liveEntryServerNodes = $deliveryProfile->getSortedLiveEntryServerNodes($entry->getEntryId(), array(EntryServerNodeStatus::PLAYABLE));
+		if(!count($liveEntryServerNodes))
+		{
+			return array();
+		}
+
+		$webVTTStreamFlavorIds = self::getWebVTTStreamFlavorIds($liveEntryServerNodes[0]);
+		if(!count($webVTTStreamFlavorIds))
+		{
+			return array();
+		}
+
+		$liveCaptions = array();
+		foreach ($entryStreams as $stream)
+		{
+		    /* @var $stream kStreamContainer */
+		    if (in_array($stream->getId(), $webVTTStreamFlavorIds))
+		    {
+				$caption = [
+				'tokenizer'=> $deliveryProfile->getTokenizer(),
+				'urlPrefix'=> $deliveryProfile->getPackagerUrl($liveEntryServerNodes[0]),
```

Comment:
i would consider adding empty function in the parent class of delivery profile since it was only implemented for live delivery profile